edge-modules/edge-dashboard/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+            edge_config,
+            settings,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Module {
+    name: String,
+    status: String,
+}
+
+impl Module {
+    pub fn new(name: String, status: String) -> Self {
+        Module {
+            name,
+            status,
+        }
+    }
+}
+
+pub struct Main {
+    context: Arc<Context>,
+}
+
+impl Main {
+    pub fn new(context: Context) -> Self {
+        Main {
+            context: Arc::new(context),
+        }
+    }
+
+    pub fn run(&self) -> Result<(), Error> {
+        let address = format!(
+            "{}:{}",
+            self.context.settings.host, self.context.settings.port
+        );
+
+        println!("Server listening at http://{}", address);
+
+        let context = web::Data::new(self.context.clone());
+        let device = web::Data::new(set_up(context.clone()));
+
+        HttpServer::new(move || {
+            App::new()
+                .register_data(context.clone())
+                .register_data(device.clone())
+                .service(web::resource("/api/modules/").to_async(get_modules))
+                .service(web::resource("/api/provisioning-state/").to(get_state))
+                .service(web::resource("/api/connectivity/").to(get_connectivity))
+        })
+        .bind(address)?
+        .run()?;
+
+        Ok(())
+    }
+}
+
+#[derive(Deserialize)]
+pub struct AuthRequest {
+    version: String,
+}
+
+fn set_up(context: web::Data<Arc<Context>>) -> Option<state::Device> {
+    if let Ok(_) = state::get_file() {
+        // if file exists and can be located
+        context.get_ref()
+            .edge_config
+            .as_ref()
+            .map(|config| {
+                if let Provisioning::Manual(val) = config.provisioning() {
+                    let dev_conn_str = val.device_connection_string();
+                    if dev_conn_str == edgelet_core::DEFAULT_CONNECTION_STRING {
+                        Some(state::Device::new(
+                            state::State::NotProvisioned,
+                            String::new(),
+                        ))
+                    } else {
+                        Some(state::Device::new(
+                            state::State::Manual,
+                            dev_conn_str.to_string(),
+                        ))
+                    }
+                } else {
+                    // handle non-manual (dps/external) here
+                    Some(state::Device::new(state::State::Manual, String::new()))
+                }
+            })
+            .unwrap_or_else(|e| {
+                println!("Err: {:?}", e);
+                None
+            })
+    } else {
+        // if file doesn't exist or can't be located
+        Some(state::Device::new(
+            state::State::NotInstalled,
+            String::new(),
+        ))
+    }
+}
+
+fn get_state(
+    _req: HttpRequest,
+    device: web::Data<Option<state::Device>>,
+    // _info: web::Query<AuthRequest>,
```

Comment:
In general, commented code should be removed before we merge.  Pleas find& remove the commented code.